src/main/scala/org/neo4j/spark/reader/Neo4jInputPartitionReader.scala

```diff
@@ -15,27 +15,37 @@ import scala.collection.JavaConverters._
 class Neo4jInputPartitionReader(private val options: Neo4jOptions,
                                 private val filters: Array[Filter],
                                 private val schema: StructType,
-                                private val jobId: String) extends InputPartition[InternalRow]
+                                private val jobId: String,
+                                private val partitionId: Int = 0,
+                                private val skip: Long = -1L,
+                                private val limit: Long = -1L) extends InputPartition[InternalRow]
   with InputPartitionReader[InternalRow]
   with Logging {
 
   private var result: Iterator[Record] = _
   private var session: Session = _
   private var transaction: Transaction = _
-  private val driverCache: DriverCache = new DriverCache(options.connection, jobId)
+  private val driverCache: DriverCache = new DriverCache(options.connection,
```

Comment:
why do we need the partitionid in the driver cache?

Reply:
Because otherwise in a multi-partition env the first one that closes the driver cache close it for everyone.